### PR TITLE
Support n color themes in StyleDictionary custom formatter

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

The style dictionary custom formatter for generating color tokens has been refactored to support additional color modes beyond `light` and `dark`. This is in preparation to upstream the formatter to Primer Primitives. 

## List of notable changes:

- **updated** `color-mode-attributes.js` file to handle new color modes
- **removed** duplication of light and root by comma separating the selectors (see screenshot below) 
- **added** new test coverage to ensure there are no regressions going forward and cover edge cases

## Steps to test:

1. Check the CI diff comment below and ensure there are no design token changes 
1. Check the new unit tests pass

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

There is now an `allowList` of supported modes. 
Screenshot 2022-09-13 at 11 30 22

**Primer Brand only requires 'dark' and 'light', but as we don't provide values for these in our property files, they won't get built.**

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

```json
{
  "brand": {
    "InlineLink": {
      "color": {
        "rest": {
          "value": "var(--base-color-scale-blue-5)",
          "dark": "var(--base-color-scale-blue-4)",
          "dark_dimmed": "var(--base-color-scale-blue-6)"
        },
        "pressed": {
          "value": "var(--base-color-scale-blue-7)",
          "dark": "var(--base-color-scale-blue-2)",
          "dark_dimmed": "var(--base-color-scale-blue-6)"
        }
      }
    }
  }
}
```

 </td>
<td valign="top">

```css
/**
 * Do not edit directly
 * Generated on Tue, 13 Sep 2022 10:33:30 GMT
 */

:root,
[data-color-mode="light"] {
  --brand-InlineLink-color-rest: var(--base-color-scale-blue-5);
  --brand-InlineLink-color-pressed: var(--base-color-scale-blue-7);
}

[data-color-mode="dark"] {
  --brand-InlineLink-color-rest: var(--base-color-scale-blue-4);
  --brand-InlineLink-color-pressed: var(--base-color-scale-blue-2);
}

[data-color-mode="dark_dimmed"] {
  --brand-InlineLink-color-rest: var(--base-color-scale-blue-6);
  --brand-InlineLink-color-pressed: var(--base-color-scale-blue-6);
}
```

</td>
</tr>
</table>

### CODE REVIEW

File: src/formats/color-mode-attributes.js [hunk 1]
```diff
@@ -19,39 +34,80 @@ function colorModeAttributes({dictionary: origDictionary, file, options}) {
...
+       * Checks if the token object contains one of the supported modes
+       * If yes; mutate the HSL-specific token object with the converted HSL value for the alternate mode
+       */
+      for (const mode of supportedModes) {
+        if (token.hasOwnProperty(mode)) {
+          tokenWithHSL[mode] = convertToHSL(token[mode])
+        }
+      }
+
+      acc.push(tokenWithHSL)
```
Comment (lukasoppermann) [inline src/formats/color-mode-attributes.js:54]: What is the output here?

Is it:

```js
{
  value: // hsl,
  dark: // hsl for dark
}
```

If it is, is there a possibility to nest into value?

```js
{
  value: {
    light: // hsl light,
    dark: // hsl dark
  }
}
```

This would bring us closer to the w3c standard. Or should this rather be done in a separate PR?

File: src/formats/color-mode-attributes.js [hunk 2]
```diff
@@ -19,39 +34,80 @@ function colorModeAttributes({dictionary: origDictionary, file, options}) {
...
 
     dictionary.allTokens = dictionary.allTokens.reduce(reduceProperties, [])
     dictionary.allProperties = dictionary.allProperties.reduce(reduceProperties, [])
   }
 
-  const selector = options.selector ? options.selector : `:root`
   const {outputReferences} = options
   let {allTokens} = dictionary
 
+  const defaultMode = supportedModes[0]
```
Comment (lukasoppermann) [inline src/formats/color-mode-attributes.js:65]: Could we move this up just below `supportedModes` to have it together? == Easier to read and understand.

Comment (rezrah) [reply src/formats/color-mode-attributes.js]: Yes definitely, forgot to move it when I hoisted the `supportedModes` 🤦 😄

File: src/formats/color-mode-attributes.js (hunk 1 above)
Comment (rezrah) [reply src/formats/color-mode-attributes.js:55]: This is a slightly deeper refactor and unrelated to the goal of this PR.. so will defer to a future PR if that's okay. The tricky part will be disentangling the default behavior of StyleDictionary, which requires a `value` property to always be present.

Comment (rezrah) [reply src/formats/color-mode-attributes.js:55]: I like the idea though, thanks @lukasoppermann 🙇

**Review by lukasoppermann — APPROVED** [review]
